ansible_rulebook/exception.py

```diff
@@ -14,8 +14,11 @@
 
 
 class ShutdownException(Exception):
+    def __init__(self, obj):
```

Comment:
Seems that `obj` can be only a `shutdown` object. IMO using a more self describing variable name and the respective type hint is convenient.